ddtrace/internal/datadog/profiling/dd_wrapper/src/exporter.cpp

```diff
@@ -316,8 +319,15 @@ Profile::Profile(ProfileType type, unsigned int _max_nframes)
     values.resize(samplers.size());
     std::fill(values.begin(), values.end(), 0);
 
-    ddog_prof_Period default_sampler = { samplers[0], 1 }; // Mandated by pprof, but probably unused
-    ddog_profile = ddog_prof_Profile_new({ &samplers[0], samplers.size() }, &default_sampler, nullptr);
+    ddog_prof_Period default_period = { samplers[0], 1 }; // Mandated by pprof, but probably unused
+    ddog_prof_Profile_NewResult res = ddog_prof_Profile_new({ &samplers[0], samplers.size() }, &default_period, nullptr);
+
+    // Check that the profile was created properly
+    if (res.tag != DDOG_PROF_PROFILE_NEW_RESULT_OK) {
+      ddog_Error_drop(&res.err);
+      throw std::runtime_error("Could not create profile");
```

Comment:
Maybe also take the actual message from the error, to allow better debugging?

Reply:
Good call--fixed